tests/tests_wrong_provider.yml

```diff
@@ -0,0 +1,26 @@
+---
+- name: Test issuing certificate with nonexistent provider
+  hosts: all
+  become: true
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: www.example.com
+        ca: self-sign
+        provider: fake-provider
+
+  tasks:
+    - vars:
+        expected_error_msg: >-
+          Ensure certificate requests Chosen provider
+           'fake-provider' is not available.
+      block:
+        - import_role:
+            name: linux-system-roles.certificate
+        - fail:
+            msg: "Certificate issued with nonexistent provider 'fake-provider'."
+      rescue:
+        - name:
+          assert:
+            that:
+              - ansible_failed_task.name == "Ensure certificate requests"
```

Comment:
`ansible_failed_task` is currently broken - https://github.com/ansible/ansible/issues/57399

See https://github.com/linux-system-roles/logging/blob/master/tests/tests_files_elasticsearch_certs_incomplete.yml#L24 for the current best practices idiom of testing for a failure with `block/rescue` by using a dummy `name: unreachable task` and `ansible_failed_result`

Reply:
Updated in b8a4c36c7b30d2da633b06937ce862c674cc0c91